internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -285,7 +297,19 @@ func (connector *SingleUseConnector) waitForServerConnection(
 		// Since the communication between the client and Secretless must be unencrypted,
 		// we fool the client into thinking that it's talking to a server that does not
 		// support encryption.
-		preloginResponse[mssql.PreloginENCRYPTION] = []byte{mssql.EncryptNotSup}
+
+		// Conditionally carry out login encryption based on what the client says
```

Comment:
This comment says `Conditionally carry out login encryption`, but it seems the login encryption always happens (considering the variable name `happyWithOnlyLoginEncryption`, and the call to ServerUpgradeToTLSForLogin() when the client selects encryption off.